kernel-impl/src/main/java/org/sakaiproject/user/impl/BaseUserDirectoryService.java

```diff
@@ -1973,6 +1971,39 @@ public UserEdit newUser(String eid)
 		return u;
 	}
 
+	public void updateUserId(String id,String newEmail)
+	{
+		try {
+			List<String> locksSucceeded = new ArrayList<String>();
+			// own or any
+			List<String> locks = new ArrayList<String>();
+			locks.add(SECURE_UPDATE_USER_ANY);
+			locks.add(SECURE_UPDATE_USER_OWN);
+			locks.add(SECURE_UPDATE_USER_OWN_EMAIL);
+			locksSucceeded = unlock(locks, userReference(id));
+
+			if(locksSucceeded.contains(SECURE_UPDATE_USER_ANY) || locksSucceeded.contains(SECURE_UPDATE_USER_OWN_EMAIL )|| locksSucceeded.contains(SECURE_UPDATE_USER_OWN)) {
+				UserEdit user = m_storage.edit(id);
+				if (user == null) {
+					M_log.error("Unexpected error when trying to validate the user");
+					return;
+				}
+				user.setEid(newEmail);
+				user.setEmail(newEmail);
+				commitEdit(user);
+			}
+			else {
+				M_log.error("User edit did not pass permission checks");
+				return;
+			}
+		} catch (UserAlreadyDefinedException e) {
+			M_log.error(e.getStackTrace());
```

Comment:
Sorry, I should have said Exception, not stack trace. So have a log line like: `log.error("A users already exists with EID of: "+newEmail, e)`